File: Python/reservation.py
class Reservation:
    def __init__(self, customer, room, check_in_date, check_out_date):
        self.customer = customer
        self.room = room
        self.check_in_date = check_in_date
        self.check_out_date = check_out_date

    def get_customer(self):
        return self.customer

    def get_room(self):
        return self.room

    def get_check_in_date(self):
        return self.check_in_date

    def get_check_out_date(self):
        return self.check_out_date

File: C++/Reservation.cpp
#include "Reservation.h"

Reservation::Reservation(Customer customer, Room room, const std::string &checkInDate, const std::string &checkOutDate)
    : customer(customer), room(room), checkInDate(checkInDate), checkOutDate(checkOutDate)
{
}

const Customer &Reservation::getCustomer() const
{
    return customer;
}

const Room &Reservation::getRoom() const
{
    return room;
}

const std::string &Reservation::getCheckInDate() const
{
    return checkInDate;
}

const std::string &Reservation::getCheckOutDate() const
{
    return checkOutDate;
}